cpp/src/arrow/csv/writer.cc

```diff
@@ -80,17 +80,19 @@ int64_t CountQuotes(util::string_view s) {
 // Matching quote pair character length.
 constexpr int64_t kQuoteCount = 2;
 constexpr int64_t kQuoteDelimiterCount = kQuoteCount + /*end_char*/ 1;
-static const std::string& str_comma = ",";
+constexpr char const* kStrComma = ",";
 
 // Interface for generating CSV data per column.
 // The intended usage is to iteratively call UpdateRowLengths for a column and
 // then PopulateColumns. PopulateColumns must be called in the reverse order of the
 // populators (it populates data backwards).
 class ColumnPopulator {
  public:
-  ColumnPopulator(MemoryPool* pool, const std::string& end_chars,
+  ColumnPopulator(MemoryPool* pool, const std::string end_chars,
                   std::shared_ptr<Buffer> null_string)
-      : end_chars_(end_chars), null_string_(std::move(null_string)), pool_(pool) {}
+      : end_chars_(std::move(end_chars)),
```

Comment:
Should remove the `const`, otherwise there will be data copying.
But first, why removing the reference?

Reply:
Wouldn't it copy end_chars anyway even if const is removed? Does having const or removing const make any difference?

Yes. The lifetime of end_chars could end early, so as the lifetime of options?

https://github.com/apache/arrow/blob/ec62a45fd65104191385e58d308d6b3c027b7a4c/cpp/src/arrow/csv/writer.cc#L406

Reply:
Here's a simple test code to compare moving an object with&without `const`: https://onlinegdb.com/tnYfwgQ5s

Reply:
Thanks. I'll remove const. Found an article that summarizes all.

https://www.nextptr.com/tutorial/ta1211389378/beware-of-using-stdmove-on-a-const-lvalue

Reply:
@cyb70289 I am getting some weird outputs with strings.

https://www.onlinegdb.com/fork/tnYfwgQ5s

va = 0x7ffead064240
A(no const): 0x7ffead0642c0, 0x7ffead064260
vb = 0x7ffead064280
B(const): 0x7ffead0642c0, 0x7ffead0642a0

Reply:
std::string has may tricks to improve performance, e.g., short string optimization (SSO) to allocate short strings directly inside the object, not in a separated buffer. So short strings are not movable.
You will see expected behaviour if you print string.size(). If it becomes 0, the string is moved, though the underlying buffer may be not freed in case it may be used again soon.

Reply:
I had to look it up.  You are seeing the effects of "short string mode": https://joellaity.com/2020/01/31/string.html

If strings are short enough they are basically stack allocated.  There is no real difference between a move and a copy of a short string.

If you make your strings longer than 22 characters then you will see expected results.

This means, in this case, the moves are not really necessary, but still it is nice to keep them for consistency's sake I think.

Reply:
Thanks, @cyb70289  and @westonpace